tg_bot/modules/log_channel.py

```diff
@@ -61,16 +61,32 @@ def log_resource(bot: Bot, update: Update):
 
             for descriptor, entity in entities.items():
                 result = f'<b>Risorsa inviata da @{update.effective_user.username}:</b>\n'
+
                 if descriptor['type'] == MessageEntity.URL:
-                    extracted = tldextract.extract(entity)
-                    if f'#{extracted.domain}' not in tags:
-                        tags.append(f'#{extracted.domain}')
-                        tags.sort()
-                if descriptor['type'] in [MessageEntity.URL, MessageEntity.TEXT_LINK]:
                     try:
                         response = requests.get(entity)
                         if response.status_code == requests.codes.ok:
                             result += f'{entity}'
+                            extracted = tldextract.extract(entity)
```

Comment:
Questa parte di estrazione del dominio e costruzione dei tags viene ripetuta più volte per le varie casistiche di MessageEntity, che ne pensi di provare a generalizzare quel flusso in qualche funzione di util così eviteresti un po' di codice ripetuto.